storage/rocksdb/ha_rocksdb.cc

```diff
@@ -1539,8 +1539,7 @@ class Rdb_transaction
                                                 false /* just statement */);
       return HA_ERR_LOCK_DEADLOCK;
     }
-
-    if (s.IsBusy())
+    else if (s.IsBusy())
```

Comment:
I wanted to use else to make sure it would never treat deadlock as snapshot conflict. Better fix would be introducing a function like IsSnapshotConflict() in rocksdb.